### ISSUE TITLE
[DOC] Guide for setting up Azure SCIM integration

### ISSUE THREAD

**Issue by dvasdekis** [databricks/terraform-provider-databricks#1917]
We have a fairly standard setup in Azure, where AD-authenticated users in our organisation belong to Azure groups, and we would like to have those users in specific groups automatically provisioned in Databricks. Specifically, we would like to have all users in specific groups to get added to the "users" group in Databricks, either when they get added to the AD group or when they first log in to Databricks.

As  mentioned [here](https://stackoverflow.com/a/74491584), SCIM integration is possible, and a guide exists to [creating it manually](https://docs.databricks.com/administration-guide/users-groups/scim/aad.html), but I would like to automate the SCIM provisioning via Terraform. We have many workspaces (50+) and many user groups (150+) so even clicking on things at our scale isn't possible. However, there's no clear guide that I can find for doing this. There are also a confusing number of permissions resources, and I don't understand how [this feature](https://github.com/databricks/terraform-provider-databricks/issues/28) relates to the SCIM provider.

Thank you!

### Expected Details
Create a guide for SCIM automation in Azure using Terraform.

### List of things to potentially add/remove:
This is a list of things to manipulate in the docs:
- [ ] Guide to SCIM provisioning in Azure

### Important Factoids
As a note, prior to the "Enable admin protection for 'No isolation shared' clusters" security feature, we were able to give these users 'Contributor' access, which would automatically give them membership of the 'admins' group in Databricks on first login. Now they are blocked from accessing `dbutils.secrets.get` if they're a member of the Admin group, and we need them to be in the 'users' group instead.

### References
#28

**Comment by TakeshiMatsukura** [comment]
Most items for SCIM provisioning are configured on AAD side not Databricks side.
So I'm not sure it will be supported on Databricks provider.

Does this help your use case?
https://stackoverflow.com/questions/73125274/how-to-configure-scim-provisioning-for-azure-ad-and-databricks-via-terraform

**Comment by dvasdekis** [comment]
> Most items for SCIM provisioning are configured on AAD side not Databricks side. So I'm not sure it will be supported on Databricks provider.

It would still be very helpful to have a guide.
 
> Does this help your use case? https://stackoverflow.com/questions/73125274/how-to-configure-scim-provisioning-for-azure-ad-and-databricks-via-terraform

Alex has done great work here, but (by my understanding) the linked script iterates through users within groups, and provisions those users in Databricks on 'terraform apply', rather than provisioning the groups and allowing new users in those groups to have access on first login.

**Comment by schmocker** [comment]
Maybe this helps: https://medium.com/@vvimal44/streamlining-your-journey-automating-scim-configuration-for-azure-databricks-with-terraform-e1a1dec21393

**Comment by alexott** [comment]
Here is TF-based sync: https://github.com/grusin-db/uc-azure-account-scim-sync

**Comment by dvasdekis** [comment]
> Maybe this helps: https://medium.com/@vvimal44/streamlining-your-journey-automating-scim-configuration-for-azure-databricks-with-terraform-e1a1dec21393

This would ordinarily be fantastic, but we're in a very large organisation which doesn't give us the ability to define `azuread_synchronization_secret` jobs. We're hoping to use an azure-native setup which doesn't require changes to active directory. Thank you though!

**Comment by schmocker** [comment]
Indeed, one would hope that with a product called **Azure** Databricks this comes out of the box 😄

**Comment by abij** [comment]
I'm a little late, but I figured it out and wrote a blog for as long as Automatic Identity Management is not GA...

https://xebia.com/blog/manage-azure-databricks-scim-with-terraform/

For your use-case: assigning new group as users to workspace, a subsequent step in TF can be added to add the group into the Workspace as users:

```terraform
resource "databricks_mws_permission_assignment" "group_is_user" {
  workspace_id = databricks_mws_workspaces.this.workspace_id
  principal_id =  group_id # reference to the group new account-level group
  permissions  = ["USER"] # or "ADMIN"
}
```
[Official Databricks TF: mws_permission_assignment](https://registry.terraform.io/providers/databricks/databricks/latest/docs/resources/mws_permission_assignment#example-usage)
